### ISSUE TITLE
Fix compiler warning of newer compilers

### ISSUE THREAD

**Issue by jxsl13** [teeworlds/teeworlds#2188]
example compiler warning: 

```
src/game/server/entities/character.cpp:580:56: warning: implicitly-declared ‘constexpr vector2_base<float>::vector2_base(const vector2_base<float>&)’ is deprecated [-Wdeprecated-copy]
  580 |    GameServer()->CreateSound(m_Pos, SOUND_WEAPON_NOAMMO);
      |                                                       
```

fix:

explicitly declare a copy constructor for vector2_base:
src/base/vmath.h:

add a copy constructor:

```
constexpr vector2_base(const vector2_base<T>& other)
	{
		x = other.x;
		y = other.y;
	}
```

**Comment by oy** [comment]
Wasn't that fixed already 7ae55cdd44e28528a2f7a66fc791f54414ea8a48 ?

**Comment by jxsl13** [comment]
sorry, true.
I'm might be on an older codebase from the tw master.
